### PR TITLE
create separate event emission for ibc software upgrades vs an upgraded client

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #4507 

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/02-client/keeper/keeper.go [hunk 1]
```diff
@@ -459,8 +459,8 @@ func (k Keeper) ScheduleIBCSoftwareUpgrade(ctx sdk.Context, plan upgradetypes.Pl
 		return err
 	}
 
-	// emitting an event for handling client upgrade proposal
-	emitUpgradeClientProposalEvent(ctx, plan.Name, plan.Height)
+	// emitting an event for scheduling an upgrade plan
+	emitScheduleIBCSoftwareUpgradeEvent(ctx, plan.Name, plan.Height, cs)
```
Comment (colin-axner) [inline modules/core/02-client/keeper/keeper.go:463]: I'd be curious to know if hermes uses the existing event type. I forget if we checked on this?

File: modules/core/02-client/types/events.go [hunk 2]
```diff
@@ -21,14 +21,15 @@ const (
...
-	EventTypeUpgradeClientProposal = "upgrade_client_proposal"
-	EventTypeRecoverClient         = "recover_client"
+	EventTypeCreateClient               = "create_client"
+	EventTypeUpdateClient               = "update_client"
+	EventTypeUpgradeClient              = "upgrade_client"
+	EventTypeSubmitMisbehaviour         = "client_misbehaviour"
+	EventTypeUpdateClientProposal       = "update_client_proposal"
+	EventTypeUpgradeChain               = "upgrade_chain"
+	EventTypeUpgradeClientProposal      = "upgrade_client_proposal"
+	EventTypeScheduleIBCSoftwareUpgrade = "schedule_ibc_client_upgrade"
```
Comment (colin-axner) [inline modules/core/02-client/types/events.go:31]: should the value be `schedule_ibc_software_upgrade`?

**Review by colin-axner — APPROVED** [review]

File: modules/core/02-client/keeper/keeper.go (hunk 1 above)
Comment (colin-axner) [reply modules/core/02-client/keeper/keeper.go:463]: Maybe we will find out if the e2e tries to use hermes to complete the upgrade

File: modules/core/02-client/types/events.go (hunk 2 above)
Comment (charleenfei) [reply modules/core/02-client/types/events.go:31]: i thought about this, but actually we are scheduling an upgrade of a client state. but happy to change it to `schedule_ibc_software_upgrade` for consistency with the method name, if that makes more sense!

File: modules/core/02-client/keeper/keeper.go (hunk 1 above)
Comment (charleenfei) [reply modules/core/02-client/keeper/keeper.go:463]: hermes just uses the `UpgradeClient` event, as i can see, not the `UpgradeClientProposal` event.

Comment (colin-axner) [reply modules/core/02-client/keeper/keeper.go:463]: Yea same here. I think they'd likely use `UpgradeChain` event instead as well

**Review by crodriguezvega — APPROVED** [review]

**Pull request merged**